File: backend/app/schemas/application.py
import uuid

from pydantic import BaseModel, Field

from app.models.application import ApplicationStatus, DocumentType
from app.schemas.common import UtcDateTime


class ApplicationDocumentRead(BaseModel):
    id: uuid.UUID
    doc_type: DocumentType
    created_at: UtcDateTime

    model_config = {"from_attributes": True}


class DriverSummary(BaseModel):
    email: str
    nic: str

    model_config = {"from_attributes": True}


class ApplicationRead(BaseModel):
    id: uuid.UUID
    driver_id: uuid.UUID
    driver: DriverSummary
    status: ApplicationStatus
    reason: str | None
    created_at: UtcDateTime
    updated_at: UtcDateTime
    documents: list[ApplicationDocumentRead]

    model_config = {"from_attributes": True}


class RejectApplicationRequest(BaseModel):
    reason: str = Field(min_length=1, max_length=1000)

File: mobile/src/types/application.ts
export type ApplicationStatus = 'PENDING' | 'APPROVED' | 'REJECTED';
export type DocumentType = 'FACE_PHOTO' | 'NIC' | 'MEDICAL_CERT' | 'BIRTH_CERT';

export interface ApplicationDocumentRead {
  id: string;
  doc_type: DocumentType;
  created_at: string;
}

export interface Application {
  id: string;
  driver_id: string;
  status: ApplicationStatus;
  reason: string | null;
  created_at: string;
  updated_at: string;
  documents: ApplicationDocumentRead[];
}
